In [1]:
import pandas as pd

In [2]:
df=pd.read_csv('ds_salaries.csv')

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 607 entries, 0 to 606
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   work_year           607 non-null    int64 
 1   experience_level    607 non-null    object
 2   employment_type     607 non-null    object
 3   job_title           607 non-null    object
 4   salary              607 non-null    int64 
 5   salary_currency     607 non-null    object
 6   salary_in_usd       607 non-null    int64 
 7   employee_residence  607 non-null    object
 8   remote_ratio        607 non-null    int64 
 9   company_location    607 non-null    object
 10  company_size        607 non-null    object
dtypes: int64(4), object(7)
memory usage: 52.3+ KB


In [10]:
df.isnull().sum()

work_year             0
experience_level      0
employment_type       0
job_title             0
salary                0
salary_currency       0
salary_in_usd         0
employee_residence    0
remote_ratio          0
company_location      0
company_size          0
dtype: int64

- The Contains of 607 recored , and no missing values were found 

In [11]:
df.describe()

work_year        salary  salary_in_usd  remote_ratio
count   607.000000  6.070000e+02     607.000000     607.00000
mean   2021.405272  3.240001e+05  112297.869852      70.92257
std       0.692133  1.544357e+06   70957.259411      40.70913
min    2020.000000  4.000000e+03    2859.000000       0.00000
25%    2021.000000  7.000000e+04   62726.000000      50.00000
50%    2022.000000  1.150000e+05  101570.000000     100.00000
75%    2022.000000  1.650000e+05  150000.000000     100.00000
max    2022.000000  3.040000e+07  600000.000000     100.00000

- Most Of the Job Salaries are around 112297.869852
- highest slary is 600000$
- lowest salary is 2859


In [18]:
df[(df['experience_level']=='SE')]['job_title']

1      Machine Learning Scientist
2               Big Data Engineer
4       Machine Learning Engineer
6             Lead Data Scientist
9              Lead Data Engineer
                  ...            
597                  Data Analyst
602                 Data Engineer
603                 Data Engineer
604                  Data Analyst
605                  Data Analyst
Name: job_title, Length: 280, dtype: object

In [20]:
df['experience_level'].value_counts()

SE    280
MI    213
EN     88
EX     26
Name: experience_level, dtype: int64

In [24]:
df.sort_values(by='salary_in_usd',ascending=False)

work_year experience_level employment_type  \
252       2021               EX              FT   
97        2021               MI              FT   
33        2020               MI              FT   
157       2021               MI              FT   
225       2021               EX              CT   
..         ...              ...             ...   
179       2021               MI              FT   
77        2021               MI              PT   
238       2021               EN              FT   
185       2021               MI              FT   
176       2021               MI              FT   

                              job_title  salary salary_currency  \
252             Principal Data Engineer  600000             USD   
97               Financial Data Analyst  450000             USD   
33                   Research Scientist  450000             USD   
157  Applied Machine Learning Scientist  423000             USD   
225            Principal Data Scientist  416000             USD   
..                                  ...     ...             ...   
179                      Data Scientist  420000             INR   
77        3D Computer Vision Researcher  400000             INR   
238                      Data Scientist    4000             USD   
185                       Data Engineer    4000             USD   
176                      Data Scientist   58000             MXN   

     salary_in_usd employee_residence  remote_ratio company_location  \
252         600000                 US           100               US   
97          450000                 US           100               US   
33          450000                 US             0               US   
157         423000                 US            50               US   
225         416000                 US           100               US   
..             ...                ...           ...              ...   
179           5679                 IN           100               US   
77            5409                 IN            50               IN   
238           4000                 VN             0               VN   
185           4000                 IR           100               IR   
176           2859                 MX             0               MX   

    company_size  
252            L  
97             L  
33             M  
157            L  
225            S  
..           ...  
179            S  
77             M  
238            M  
185            M  
176            S  

[607 rows x 11 columns]